### PR TITLE
Initial commit of an empty Angular app

### PR DESCRIPTION
**Pull request by razvanm**
Angular was installed in the root of this repo using the following
command:

    npm install @angular/cli

The Angular app was created using the following command and answers:

    ng new web
    ? Would you like to add Angular routing? Yes
    ? Which stylesheet format would you like to use? SCSS   [ https://sass-lang.com/documentation/syntax#scss ]

The app can be run locally from the web/ directory by running:

  npm run start

### CODE REVIEW

File: web/tslint.json [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    "no-outputs-metadata-property": true,
+    "template-banana-in-box": true,
+    "template-no-negated-async": true,
+    "use-lifecycle-interface": true,
+    "use-pipe-transform-interface": true
+  },
+  "rulesDirectory": [
+    "codelyzer"
+  ]
+}
```
Comment (carolksun) [inline web/tslint.json:152]: missing new line at end of file

File: web/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  To learn more about this file see: https://angular.io/config/solution-tsconfig.
+*/
+{
+  "files": [],
+  "references": [
+    {
+      "path": "./tsconfig.app.json"
+    },
+    {
```
Comment (LujieZ) [inline web/tsconfig.json:13]: Maybe combine these two lines?

File: web/src/polyfills.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ *  (window as any).__Zone_enable_cross_context_check = true;
+ *
+ */
+
+/***************************************************************************************************
+ * Zone JS is required by default for Angular itself.
+ */
+import 'zone.js/dist/zone';  // Included with Angular CLI.
+
+
```
Comment (LujieZ) [inline web/src/polyfills.ts:60]: An extra line here.

File: web/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+      "path": "./tsconfig.spec.json"
+    }
+]
+}
```
Comment (LujieZ) [inline web/tsconfig.json:17]: Indentation error here.

File: web/src/polyfills.ts (hunk 3 above)
Comment (razvanm) [reply web/src/polyfills.ts:60]: I think the extra empty line makes sense here to better separate the different sections in this file. Note that this file was generated by the `ng new` command. :-)

File: web/.editorconfig [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+indent_size = 2
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.ts]
+quote_type = single
+
+[*.md]
+max_line_length = off
+trim_trailing_whitespace = false
```
Comment (yinggao) [inline web/.editorconfig:16]: Shall we also add `insert_final_newline = false` for .md files?

File: web/.gitignore [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+# IDE - VSCode
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
+.history/*
+
+# misc
```
Comment (yinggao) [inline web/.gitignore:34]:
```suggestion
# misc.
```

File: web/.editorconfig (hunk 5 above)
Comment (razvanm) [reply web/.editorconfig:16]: Why do we want `insert_final_newline` set to `false`?

File: web/.gitignore (hunk 6 above)
Comment (razvanm) [reply web/.gitignore:34]: This file was autogenerated by `ng new` so I would like to let it as is in this PR.

File: web/.editorconfig (hunk 5 above)
Comment (yinggao) [reply web/.editorconfig:16]: Ah, I think I miss read it.

File: web/.gitignore (hunk 6 above)
Comment (yinggao) [reply web/.gitignore:34]: Got it.

**Comment by razvanm** [comment]
I removed the `env` dir and the top level `package-lock.json` because the `ng` binary available via `npm run ng` from the `web/` directory.

Please take one last look. :-)

**Comment by razvanm** [comment]
> I saw the error message " Your tests failed on CircleCI", should we fix it?

The failure is due to lack of the CircleCI config file. #5 is the PR that adds that. So the failure is expected. :-)

**Pull request merged**